cpp/bridges_module/bridges_test.cpp

```diff
@@ -0,0 +1,211 @@
+  //    (2)     (6)
+  auto graph = mg_generate::BuildGraph(8, {{0, 1}, {0, 2}, {1, 2}, {1, 3}, {3, 4}, {3, 7}, {4, 5}, {4, 6}, {5, 6}});
+  auto bridges = bridges_alg::GetBridges(*graph);
+  ASSERT_TRUE(CheckBridges(bridges, {{1, 3}, {3, 4}, {3, 7}}));
+}
+
+TEST(Bridges, HandmadeConnectedGraph2) {
+  //    (1)--(3)--(7)     (10)     (14)
+  //   / |    |    |     /    \     |
+  // (0) |   (4)--(5)--(9)   (12)--(13)
+  //   \ |     \  / \    \   /
+  //    (2)     (6) (8)   (11)
+  auto graph = mg_generate::BuildGraph(15, {{0, 1},
+                                            {0, 2},
+                                            {1, 2},
+                                            {1, 3},
+                                            {3, 4},
+                                            {3, 7},
+                                            {4, 5},
+                                            {4, 6},
+                                            {5, 6},
+                                            {5, 7},
+                                            {5, 8},
+                                            {5, 9},
+                                            {9, 10},
+                                            {9, 11},
+                                            {10, 12},
+                                            {11, 12},
+                                            {12, 13},
+                                            {13, 14}});
+  auto bridges = bridges_alg::GetBridges(*graph);
+  ASSERT_TRUE(CheckBridges(bridges, {{1, 3}, {5, 8}, {5, 9}, {12, 13}, {13, 14}}));
+}
+
+TEST(Bridges, HandmadeDisconnectedGraph) {
+  //    (4)--(5)                   (12)         (19)         (23)
+  //     |    |                   /    \        /           /   \
+  //    (1)--(3)               (11)    (13)--(18)--(20)--(22)--(24)
+  //   /                        |       |       \
+  // (0)             (8)--(9)  (10)----(14)      (21)--(25)
+  //   \                        |
+  //    (2)--(6)               (15)----(16)
+  //      \  /                    \    /
+  //       (7)                     (17)
+  auto graph = mg_generate::BuildGraph(
+      26, {{0, 1},   {0, 2},   {1, 4},   {1, 3},   {3, 5},   {4, 5},   {2, 6},   {2, 7},   {6, 7},   {8, 9},
+           {10, 11}, {11, 12}, {12, 13}, {13, 14}, {10, 14}, {10, 15}, {15, 16}, {16, 17}, {15, 17}, {13, 18},
+           {18, 19}, {18, 21}, {18, 20}, {21, 25}, {20, 22}, {22, 23}, {23, 24}, {22, 24}});
+  auto bridges = bridges_alg::GetBridges(*graph);
+  ASSERT_TRUE(CheckBridges(
+      bridges, {{0, 1}, {0, 2}, {8, 9}, {10, 15}, {13, 18}, {18, 19}, {18, 20}, {18, 21}, {21, 25}, {20, 22}}));
+}
+
+TEST(Bridges, SimpleNeighborCycle) {
+  auto graph = mg_generate::BuildGraph(2, {{0, 1}, {1, 0}});
+  auto bridges = bridges_alg::GetBridges(*graph);
+  ASSERT_EQ(0, bridges.size());
+}
+
+TEST(Bridges, NeighborCycle) {
+  auto graph = mg_generate::BuildGraph(6, {{0, 1}, {1, 2}, {2, 3}, {3, 0}, {3, 4}, {3, 4}, {4, 5}});
```

Comment:
Why do we duplicate `{3,4}` edge?

Reply:
This is a wild catch. Although there is no special reason for this, rather that it is a mistake made by the original code author.

Reply:
Update: seems like this test was checking a neighboring cycle, meaning two vertices connected with two edges. This shouldn't result as bridge in the algorithm, meaning that code was correct.